src/highdicom/content.py

```diff
@@ -936,3 +1067,536 @@ def __init__(
+
+        if voi_luts is not None:
+            if len(voi_luts) == 0:
+                raise ValueError('"voi_luts" should not be empty.')
+            for lut in voi_luts:
+                if not isinstance(lut, LUT):
+                    raise TypeError(
+                        'Items of "voi_luts" should be of type highdicom.LUT.'
+                    )
+            self.VOILUTSequence = list(voi_luts)
+        else:
+            if window_center is None:
+                raise TypeError(
+                    'At least one of "window_center" or "voi_luts" should be '
+                    'provided.'
+                )
+
+
+class PaletteColorLookupTable(Dataset):
+
+    """Dataset describing a palette color lookup table."""
+
+    def __init__(
+        self,
+        red_palette_color_lut_data: np.ndarray,
+        green_palette_color_lut_data: np.ndarray,
+        blue_palette_color_lut_data: np.ndarray,
+        red_first_mapped_value: int,
+        green_first_mapped_value: int,
+        blue_first_mapped_value: int,
+        palette_color_lut_uid: Union[UID, str, None] = None
+    ):
+        """Add attributes from the Palette Color Lookup Table module.
+
+        Parameters
+        ----------
+        red_palette_color_lut_data: np.ndarray
+            Array of values for the red color lookup table data. Must be a 1D
+            array of uint16 values, where the first entry is the red output
+            value of the palette color lookup table operation when the input
+            pixel is ``"red_first_mapped_value"``, and so on.
+        green_palette_color_lut_data: np.ndarray
+            Array of values for the green color lookup table data. Otherwise as
+            described for ``red_palette_color_lut_data``.
+        blue_palette_color_lut_data: np.ndarray
+            Array of values for the blue color lookup table data. Otherwise as
+            described for ``red_palette_color_lut_data``.
+        red_first_mapped_value: int
+            Integer representing the first input value mapped by the red palette
+            lookup table operation.
+        green_first_mapped_value: int
+            Integer representing the first input value mapped by the green
+            lookup table operation.
+        blue_first_mapped_value: int
+            Integer representing the first input value mapped by the blue
+            palette lookup table operation.
+        palette_color_lut_uid: Union[UID, str, None], optional
+            Unique identifier for the palette color lookup table.
+
+        """
+        super().__init__()
+        colors = ['red', 'green', 'blue']
+        all_lut_data = [
+            red_palette_color_lut_data,
+            green_palette_color_lut_data,
+            blue_palette_color_lut_data
+        ]
+        all_first_values = [
+            red_first_mapped_value,
+            green_first_mapped_value,
+            blue_first_mapped_value
+        ]
```

Comment:
The bits allocated and length have to be the same and I'm I'm now checking for this. I don't believe that the first mapped values have to be the same, although it would be rather odd if they are not.

As an aside, if the descriptors did need to be the same, there would be no reason to have three separate ones...

Reply:
[Part 03 Section C.7.6.3.1.5 Palette Color Lookup Table Descriptor](https://dicom.nema.org/medical/dicom/current/output/chtml/part03/sect_C.7.6.3.html#sect_C.7.6.3.1.5):

> The first value shall be identical for each of the Red, Green, Blue and Alpha Palette Color Lookup Table Descriptors.
> The second value shall be identical for each of the Red, Green, Blue and Alpha Palette Color Lookup Table Descriptors.
> The third value shall be identical for each of the Red, Green and Blue Palette Color Lookup Table Descriptors.

Agreed, it doesn't make sense to have three separate attributes, but what shall we do...

Reply:
Huh, I think when I was reading this before I was on a slightly different page that didn't mention that the second values had to match.

Added check for the second value in a4708cf